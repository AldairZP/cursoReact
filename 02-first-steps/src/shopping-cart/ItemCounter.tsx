import { useState } from "react";

import "./ItemCounter.css";
import styles from "./ItemCounter.module.css";

export interface Props {
  name: string;
  quantity?: number;
  // quantity: number | undefined; // Tiene que ponerse el prop en la etiqueta
}

export const ItemCounter = ({ name, quantity = 1 }: Props) => {
  const [count, setCount] = useState(quantity);

  const handleAdd = () => {
    setCount((value) => value + 1);
  };

  const handleSubtract = () => {
    if (count === 1) return;
    setCount((value) => value - 1);
  };

  return (
    <section
      className={styles["item-row"]}
      // style={{
      //   display: "flex",
      //   alignItems: "center",
      //   gap: 10,
      //   marginTop: 10,
      // }}
    >
      <span
        className={styles.itemText}
        style={{
          color: count === 1 ? "red" : "black",
        }}
      >
        {name}
      </span>
      <button onClick={handleAdd}>+1</button>
      <span>{count}</span>
      <button onClick={handleSubtract}>-1</button>
    </section>
  );
};
