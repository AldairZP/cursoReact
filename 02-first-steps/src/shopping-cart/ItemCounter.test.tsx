import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { ItemCounter } from "./ItemCounter";

describe("ItemCounter", () => {
  test("should decrease count when -1 is pressed", () => {
    const name = "Test Item";
    const quantity = 2;

    render(<ItemCounter name={name} quantity={quantity} />);
    const [, buttonSubtract] = screen.getAllByRole("button");

    fireEvent.click(buttonSubtract);

    expect(screen.getByText(quantity - 1)).toBeDefined();
  });

  test("should not decrease count when -1 is pressed and quantity is 1", () => {
    const name = "Test Item";
    const quantity = 1;

    render(<ItemCounter name={name} quantity={quantity} />);
    const [, buttonSubtract] = screen.getAllByRole("button");

    fireEvent.click(buttonSubtract);

    expect(screen.getByText(1)).toBeDefined();
  });

  test("snapshot", () => {
    const name = "Test Item";
    const quantity = 3;
    const { container } = render(
      <ItemCounter name={name} quantity={quantity} />
    );

    expect(container).toMatchSnapshot();
  });
});
