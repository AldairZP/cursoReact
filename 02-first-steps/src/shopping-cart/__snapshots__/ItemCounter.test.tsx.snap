// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ItemCounter > snapshot 1`] = `
<div>
  <section
    class="_item-row_8b32a2"
  >
    <span
      class="_itemText_8b32a2"
      style="color: black;"
    >
      Test Item
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
